components/Authentication/signup.jsx

```diff
@@ -60,107 +53,155 @@ const ShowText = styled.Text`
-            break;
-
-          case 'auth/email-already-in-use':
-            alert('That email is already being used.');
-            break;
-
-          default:
-            alert(`Error: ${errorMessage}`);
-        }
-      });
+    if (name.length > 1) {
+      firebase.auth().createUserWithEmailAndPassword(email, password)
+        .then((user) => {
+          user.user.updateProfile({
+            displayName: name,
+          }).then(() => {
+            navigation.navigate('Root');
+          }).catch((error) => {
+            console.log('Display name not set.');
+            console.log(error);
+          });
+        })
+        .catch((error) => {
+          const errorMessage = error.message;
+          // TODO: this alert looks odd for some reason,
+          // can't remember what an invalid password alert looks like
+          Alert.alert('Error:', errorMessage);
+        });
+    } else {
+      alert('Please enter your name.');
+    }
   }
 
   function loginPress() {
     navigation.navigate('Login');
   }
 
-  firebase.auth().onAuthStateChanged(() => {
-    if (firebase.auth().currentUser) {
-      navigation.navigate('Root');
-    }
-  });
-
   return (
-    <KeyboardAvoidingView
-      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
-      style={{ flex: 1 }}
+    <KeyboardAwareScrollView
+      resetScrollToCoords={{ x: 0, y: 0 }}
+      contentContainerStyle={{ justifyContent: 'center', alignItems: 'center', height: '100%' }}
     >
-      <Container>
-        <Title>GainChanger</Title>
-        <SubTitle>Signup</SubTitle>
-        <Image
-          style={{ width: 250, height: 250 }}
+      <Title>GainChanger</Title>
+      <SubTitle>Signup</SubTitle>
+      <Image
+        style={{ width: 250, height: 250 }}
           /* eslint-disable global-require */
-          source={require('../../assets/logo.png')}
-          /* eslint-enable global-require */
-        />
-        <InputLine
-          placeholder="Name"
-          selectionColor="#A192FF"
-          textContentType="name"
-          value={name}
-          onChangeText={(text) => setName(text)}
-        />
+        source={require('../../assets/logo.png')}
+      />
+      <InputLine
+        placeholder="Name"
+        selectionColor="#A192FF"
+        textContentType="name"
+        autoCorrect={false}
+        value={name}
+        onChangeText={(text) => setName(text)}
+      />
+      {validName && <View style={{ paddingLeft: 15, width: '80%', flexDirection: 'row' }}><ErrorText>Name must be longer than 1 character.</ErrorText></View>}
+      {!validName && <ViewFiller />}
+      <InputLine
+        placeholder="Email"
+        selectionColor="#A192FF"
+        textContentType="emailAddress"
+        value={email}
+        onChangeText={(text) => setEmail(text)}
+      />
+      {validEmail && <View style={{ paddingLeft: 15, width: '80%', flexDirection: 'row' }}><ErrorText>Invalid email.</ErrorText></View>}
+      {!validEmail && <ViewFiller />}
```

Comment:
nit: These can be simplified into one statement with a ternary operator.

Reply:
good catch